src/Microsoft.VisualStudio.Services.Agent/ConfigurationStore.cs

```diff
@@ -46,6 +46,12 @@ public sealed class AgentSettings
 
         [DataMember(EmitDefaultValue = false)]
         public int DeploymentGroupId { get; set; }
+
+        [DataMember(EmitDefaultValue = false)]
+        public string ProjectId { get; set; }
```

Comment:
deployment group is project level and projects can be renamed. In case project rename un-configuration will fail as project got renamed. Deployment group object contains the project id, it will be available while configuration for deployment group